presto-spi/src/main/java/com/facebook/presto/spi/relation/LambdaDefinitionExpression.java

```diff
@@ -111,4 +115,55 @@ private static void checkArgument(boolean condition, String message, Object... m
             throw new IllegalArgumentException(format(message, messageArgs));
         }
     }
+    private static class CanonicalizeExpression
+            implements RowExpressionVisitor<String, Void>
+    {
+        private final Map<String, String> canonicalizedArguments = new HashMap<>();
+
+        public CanonicalizeExpression(List<String> arguments, List<Type> argumentTypes)
+        {
+            for (int i = 0; i < arguments.size(); i++) {
+                canonicalizedArguments.put(arguments.get(i), format("%s_%d", argumentTypes.get(i).toString(), i));
+            }
+        }
+
+        @Override
+        public String visitCall(CallExpression call, Void context)
+        {
+            return format("%s(%s)", call.getDisplayName(), String.join(", ", call.getArguments().stream().map(e -> e.accept(this, null)).collect(Collectors.toList())));
```

Comment:
Is `displayName` guarantee to be distinct for each function call? :) 

Also I personally like to have line breaks when using Stream API. This helps read the code from cellphone 😛 :

```java
        return format(
                "%s(%s)", 
                call.getDisplayName(), 
                String.join(", ", call.getArguments().stream()
                        .map(e -> e.accept(this, null))
                        .collect(Collectors.toList())));
```

Reply:
What do you mean distinct name for each function call? If two functions have the same name with same input they are the same function right?

Reply:
@rongrong : Right. Just wondering why it's called "display name". Is there "internal name" as well? 😛

Reply:
Hmm, good point. The display name could theoretically be different from the function name in signature but I don't think we ever do that. I think this was added to get the name of a function for "display" purpose (vs. resolve) because some call site does not have information to get function name from function handle (no access to `FunctionManager`). I don't remember the details though. So theoretically there could be a conflict but practically as long as we always pass in the same name for display name (which I believe we do) it should fine.